### PR TITLE
Formvalid: process server error messages on inline radio/checkbox

### LINKED ISSUE #9684: [BUG] Form Validation - Inline radio and checkbox inputs with wb-server-error class not recognized as errors to display
## Required information

**Describe the bug**
Using the markup for including server side errors is not recognized on inline radio and checkbox inputs.  This prevents the Error Summary from being rendered when the page loads.

**To Reproduce**
Steps to reproduce the behaviour:
See issue #9682  for an example and possible fix.

**Expected behaviour**
When the page is rendered, the Form Validation Error Summary should be displayed. The error message from the server side validation should be updated with the prefix and error counter **Error #;**.

**Who shall do the work?**
I am asking for Principal publisher to please do the work.  I don't have the tools to properly code and test a solution.

## Additional information (optional)

**Version of WET-BOEW/GCWeb you are using**
CDTS version 4.1.0

### PR DESCRIPTION
**Pull request by sgdowney**
## What does this pull request (PR) do? / Que fait cette demande « pull » (PR)?
This pull request updates the Formvalid plug-in to recognize server error messages on inline radio and checkbox inputs.

**Related issues / Requêtes associées**
Closes #9684 and #9682

### CODE REVIEW

**Comment by duboisp** [comment]
@Garneauma can you review, test and ensure this still work with CDTS or previous implementation as demoed.

**Comment by duboisp** [comment]
See #9686 for previous conversation about this change request

File: src/plugins/formvalid/formvalid-server-fr.hbs [hunk 1]
```diff
@@ -115,9 +112,36 @@
...
+						<label class="radio-inline">
+							<input class="input-validation-error" data-msg-required="Vous devez sélectionner Oui ou Non." data-rule-required="true" id="workNo" name="workStatus" type="radio" value="No" />
+							Non
+						</label>
+						<label class="radio-inline">
+							<input class="input-validation-error" id="workYes" checked="checked" name="workStatus" type="radio" value="Yes" />
+							Oui
+						</label>
+					</fieldset>
+
```
Comment (Garneauma) [inline src/plugins/formvalid/formvalid-server-fr.hbs:129]: Please add another example for `.checkbox-inline`.

```suggestion
<fieldset class="form-inline chkbxrdio-grp">
	<legend class="required"><span class="field-name">Animaux de compagnie</span> <strong class="required">(Champ requis)</strong><span class="label label-danger wb-server-error" id="petError">Un chevreuil n'est pas un animal de compagnie</span></legend>
	<label class="checkbox-inline"><input type="checkbox" checked="checked" name="petChoice" id="petDeer" required="required" /> Chevreuil</label>
	<label class="checkbox-inline"><input type="checkbox" name="petChoice" id="petCat" /> Chat</label>
	<label class="checkbox-inline"><input type="checkbox" name="petChoice" id="petDog" /> Chien</label>
</fieldset>
```

File: src/plugins/formvalid/formvalid-server-en.hbs [hunk 2]
```diff
@@ -121,7 +107,35 @@
...
+						<label class="radio-inline">
+							<input class="input-validation-error" data-msg-required="You must select Yes or No." data-rule-required="true" id="workNo" name="workStatus" type="radio" value="No" />
+							No
+						</label>
+						<label class="radio-inline">
+							<input class="input-validation-error" id="workYes" checked="checked" name="workStatus" type="radio" value="Yes" />
+							Yes
+						</label>
+					</fieldset>
+
```
Comment (Garneauma) [inline src/plugins/formvalid/formvalid-server-en.hbs:124]: Please add another example for `.checkbox-inline`.

```suggestion
<fieldset class="form-inline chkbxrdio-grp">
	<legend class="required"><span class="field-name">Owned pets</span> <strong class="required">(Required field)</strong><span class="label label-danger wb-server-error" id="petError">A deer is not a pet.</span></legend>
	<label class="checkbox-inline"><input type="checkbox" checked="checked" name="petChoice" id="petDeer" required="required" /> Deer</label>
	<label class="checkbox-inline"><input type="checkbox" name="petChoice" id="petCat" /> Cat</label>
	<label class="checkbox-inline"><input type="checkbox" name="petChoice" id="petDog" /> Dog</label>
</fieldset>
```

File: src/plugins/formvalid/formvalid-server-fr.hbs [hunk 3]
```diff
@@ -28,7 +28,7 @@
 
 	{{>alertariahidden}}
 
-	<p>Après avoir soumis votre page, le serveur retoune 4 erreurs serveur. Maintenant si vous essayez de resoumettre
+	<p>Après avoir soumis votre page, le serveur retoune 6 erreurs serveur. Maintenant si vous essayez de resoumettre
```
Comment (Garneauma) [inline src/plugins/formvalid/formvalid-server-fr.hbs:31]: Added example so we need to increase the number.

```suggestion
	<p>Après avoir soumis votre page, le serveur retourne 7 erreurs serveur. Maintenant si vous essayez de resoumettre
```

File: src/plugins/formvalid/formvalid-server-en.hbs [hunk 4]
```diff
@@ -28,7 +28,7 @@
 
 	{{>alertariahidden}}
 
-	<p>After submitting your page, the server returned 4 server errors. Now if you try to resubmit your form with some
+	<p>After submitting your page, the server returned 6 server errors. Now if you try to resubmit your form with some
```
Comment (Garneauma) [inline src/plugins/formvalid/formvalid-server-en.hbs:31]: Added example so we need to increase the number.

```suggestion
	<p>After submitting your page, the server returned 7 server errors. Now if you try to resubmit your form with some
```

File: src/plugins/formvalid/formvalid.js [hunk 5]
```diff
@@ -374,12 +374,20 @@ var componentName = "wb-frmvld",
...
+							}
+						} else if ( $( this ).next( ".radio-inline" ).children( "input" )[ 0 ] ) {
+							if ( $( this ).find( $( this ).next( ".radio-inline" ).children( "input" )[ 0 ].id ) ) {
+								$form.validate().element( $( this ).next( ".radio-inline" ).children( "input" )[ 0 ] );
+							}
+						} else if ( $( this ).next( ".label-inline" ).children( "input" )[ 0 ] ) {
+							if ( $( this ).find( $( this ).next( ".label-inline" ).children( "input" )[ 0 ].id ) ) {
+								$form.validate().element( $( this ).next( ".label-inline" ).children( "input" )[ 0 ] );
 							}
 						}
```
Comment (Garneauma) [inline src/plugins/formvalid/formvalid.js:392]: Merge those conditions.
```suggestion
						} else if ( $( this ).next( ".radio-inline, .checkbox-inline, .label-inline" ).children( "input" )[ 0 ] ) {
							if ( $( this ).find( $( this ).next( ".radio-inline, .checkbox-inline, .label-inline" ).children( "input" )[ 0 ].id ) ) {
								$form.validate().element( $( this ).next( ".radio-inline, .checkbox-inline, .label-inline" ).children( "input" )[ 0 ] );
							}
						}
```

**Review by Garneauma — CHANGES_REQUESTED** [review]
@sgdowney Thank you for your contribution! Please review the following comments.

File: src/plugins/formvalid/formvalid-server-fr.hbs (hunk 1 above)
Comment (sgdowney) [reply src/plugins/formvalid/formvalid-server-fr.hbs:129]: Example added.

File: src/plugins/formvalid/formvalid-server-en.hbs (hunk 2 above)
Comment (sgdowney) [reply src/plugins/formvalid/formvalid-server-en.hbs:124]: example added

File: src/plugins/formvalid/formvalid-server-fr.hbs (hunk 3 above)
Comment (sgdowney) [reply src/plugins/formvalid/formvalid-server-fr.hbs]: Count updated.

File: src/plugins/formvalid/formvalid-server-en.hbs (hunk 4 above)
Comment (sgdowney) [reply src/plugins/formvalid/formvalid-server-en.hbs]: Count updated.

File: src/plugins/formvalid/formvalid.js (hunk 5 above)
Comment (sgdowney) [reply src/plugins/formvalid/formvalid.js]: Condition combined.

**Review by Garneauma — APPROVED** [review]

**Comment by duboisp** [comment]
@Garneauma can you confirm what type of change (Major, Minor, Patch) this PR do?

**Pull request merged**